pyiceberg/catalog/rest/auth.py

```diff
@@ -109,6 +110,38 @@ def auth_header(self) -> str:
         return f"Bearer {self._token}"
 
 
+class GoogleAuthManager(AuthManager):
+    """
```

Comment:
I believe this is not right place for a vendor specific AuthManager but I am not sure structure of python. May be we can introduce a gcp module and collect all GCS, Catalog and AuthManager things under one specific module.